app/components/inputs/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,143 @@
+import { ImageListType, ImageType } from "react-images-uploading";
+import { useState } from "react";
+import axios from "axios";
+import { toast } from "react-hot-toast";
+import { PreviewImageWithUrl } from "@/app/types";
+
+interface Props {
+  value: PreviewImageWithUrl[];
+  onChange: (value: PreviewImageWithUrl[]) => void;
+}
+
+export const useUploadImage = ({ value: images, onChange }: Props) => {
+  const [previewImages, setPreviewImages] = useState(
+    images.map((image) => {
+      const { url, ...rest } = image;
+      return { ...rest, dataURL: url as string } as ImageType;
+    })
+  );
+
+  const [isUploading, setIsUploading] = useState(false);
+
+  const uploadImage = async (imageData: string) => {
+    const axiosResponse = await axios.post("/api/images", {
+      imageData,
+    });
+    return axiosResponse.data.url as string;
+  };
+
+  const onImageFileChange = async (
```

Comment:
no need to create async function if there's no promise lol

Reply:
There is an `await` way down